### PR TITLE
Indev 1.12.2

### PR DESCRIPTION
**Pull request by CovertJaguar**
Testing the travis app mostly.

### CODE REVIEW

**Comment by liach** [comment]
@CovertJaguar The main issue in the two recipe classes is that we cannot make sure if recipe users check matches before getting remaining items.

File: src/main/java/mods/railcraft/common/util/crafting/RemainingItemShapelessRecipe.java [hunk 1]
```diff
@@ -73,7 +74,7 @@ public ItemStack getRecipeOutput() {
 
     @Override
     public NonNullList<ItemStack> getRemainingItems(InventoryCrafting inv) {
-        if (lastResult == null) {
+        if (lastInv != inv) {
             matches(inv, null);
```
Comment (liach) [inline src/main/java/mods/railcraft/common/util/crafting/RemainingItemShapelessRecipe.java:78]: See https://pastebin.com/iK3V4RLX
https://github.com/raoulvdberge/refinedstorage/blob/ec89e43e35fd039d9aa2ae04182a2caaa97733e3/src/main/java/com/raoulvdberge/refinedstorage/apiimpl/autocrafting/CraftingPattern.java#L227

I'd assume modders aren't malicious enough to edit the grid between matches and getRemainingItems calls.

**Review by liach — APPROVED** [review]

**Comment by CovertJaguar** [comment]
hmm... I have no idea what this checks tab is supposed to look like...

**Comment by liach** [comment]
I think it is for travis ci and other things like cla

**Comment by liach** [comment]
Consider squash merging this.

**Comment by liach** [comment]
ah, travis ci indeed broke somehow

**Comment by liach** [comment]
Will get an alternative fix on the recipe.

**Comment by CovertJaguar** [comment]
More Travis testing

**Comment by liach** [comment]
@CovertJaguar Appears you need to remove the old continuous integration travis ci

**Comment by CovertJaguar** [comment]
Looks like it works!

**Comment by liach** [comment]
Hmm, should I move the fix to rebornstorage to another branch? This one has too many commits and is not always right if people modify the grid and get remainders without doing matches check. Should I make sure things are totally right at the cost of performacne?

**Comment by CovertJaguar** [comment]
I haven't even looked at the code or problem, it was just a convenient branch for testing Travis. But yes, your private branches should probably be deprecated in favor of more targeted temporary branches.

**Pull request closed without merging**